src/tokenSearch/SearchFilteringSelections.jsx

```diff
@@ -0,0 +1,98 @@
+// Lists all the exchanges and their allowed networks.
+const networksExchanged = {
+    Ethereum: ['UniswapV3', 'SushiSwap'],
+    Avalanche: ['Pangolin', 'TraderJoe'],
+    BSC: ['PancakeSwap', 'MDex']
+};
+
+// Lists all the exchanges and their allowed networks.
+const exchangesNetworked = {
+    UniswapV3: ['Ethereum'],
+    SushiSwap: ['Ethereum'],
+    Pangolin: ['Avalanche'],
+    TraderJoe: ['Avalanche'],
+    PancakeSwap: ['BSC'],
+    MDex: ['BSC']
+};
+
+/**
+ * 
+ * 
+ * @param {*} param0 
+ * @returns 
+ */
+export const SearchSelectionNetwork = ({ networkName, searchNetworks, setSearchNetworks, searchExchanges, setSearchExchanges }) => {
+    // ----------------------------------------------------------------------------------------------------------------------------------------------------
+    // Rendering.
+    return <>
+        <input
+            type="checkbox"
+            checked={searchNetworks.includes(networkName)}
+            onChange={() => {
+                // Validate that "searchNetworks" contains "networkName".
+                if (searchNetworks.includes(networkName)) {
+                    // Removes "networkName" from "searchNetworks".
+                    searchNetworks = searchNetworks.filter(network => network !== networkName);
+
+                    // Removes the exchanges associated to "networkName" since it's removed from the search.
+                    searchExchanges = searchExchanges.filter(exchance => !networksExchanged[networkName].includes(exchance));
+                }
+                else {
+                    // Adds "networkName" into "searchNetworks".
+                    searchNetworks.push(networkName);
+                }
+
+                // Sets the new value for "searchNetworks" using a "Set" to *lazy* clean duplicates; who knows?
```

Comment:
I like lodash and we have used it a lot elsewhere. 
https://lodash.com/docs/4.17.15#uniq